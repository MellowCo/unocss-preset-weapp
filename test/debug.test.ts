import { createGenerator } from '@unocss/core'
import { expect, it } from 'vitest'
import presetWeapp from '../src/index'

it('debug', async () => {
  const uno = createGenerator({
    presets: [
      presetWeapp(),
    ],
    theme: {
      colors: {
        blackA7: 'hsla(0, 0%, 0%, 0.169)',
      },
    },
  })

  expect((await uno.generate('shadow-[0_0_7.5rem_0_var(--shadow)]', { preflights: false })).css)
    .toMatchInlineSnapshot(`
      "/* layer: default */
      .shadow-_lfl_0_0_7_dl_5rem_0_var_lbl_--shadow_lbr__lfr_{--un-shadow:0 0 7.5rem 0 var(--un-shadow-color, var(--shadow));box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow);}"
    `)

  expect((await uno.generate('shadow-[0_0_7.5rem_0_#000]', { preflights: false })).css)
    .toMatchInlineSnapshot(`
      "/* layer: default */
      .shadow-_lfl_0_0_7_dl_5rem_0__wn_000_lfr_{--un-shadow:0 0 7.5rem 0 var(--un-shadow-color, rgb(0 0 0));box-shadow:var(--un-ring-offset-shadow), var(--un-ring-shadow), var(--un-shadow);}"
    `)
})
