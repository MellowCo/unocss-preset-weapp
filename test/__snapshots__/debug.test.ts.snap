// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`debug 1`] = `
"/* layer: default */
.hover__bg-green-500:hover{--un-bg-opacity:1;background-color:rgb(34 197 94 / var(--un-bg-opacity));}
.hover_a_bg-green-500:hover{--un-bg-opacity:1;background-color:rgb(34 197 94 / var(--un-bg-opacity));}
.active__bg-red-500:active{--un-bg-opacity:1;background-color:rgb(239 68 68 / var(--un-bg-opacity));}
.active_a_bg-red-500:active{--un-bg-opacity:1;background-color:rgb(239 68 68 / var(--un-bg-opacity));}"
`;
